.github/workflows/gradle.yml

```diff
@@ -28,9 +37,27 @@ jobs:
         run: chmod +x gradlew
       - name: Compile
         run: ./gradlew assemble
+
+      # this step is required for Java version 16 and up, because some internal classes used
+      # for the spotless plugin are no longer accessible. This is the official workaround provided
+      # on the google-java-format github page.
+      # these settings will cause Java 8 to crash, so they are only applied to later versions.
+      - name: Set up Gradle workaround for formatting and testing
```

Comment:
If we find ourselves repeating this, we could create our action for it xD 
https://docs.github.com/en/actions/creating-actions